stacks/serverless/dovetail-cdn-arranger.yml

```diff
@@ -0,0 +1,133 @@
+    Type: AWS::S3::Bucket
+    Properties:
+      # TODO: tag-specific expirations
+      LifecycleConfiguration:
+        Rules:
+          - ExpirationInDays: 30
+            Status: Enabled
+      Tags:
+        - Key: Project
+          Value: Dovetail
+        - Key: Environment
+          Value: !Ref EnvironmentType
+        - Key: prx:cloudformation:stack-name
+          Value: !Ref AWS::StackName
+        - Key: prx:cloudformation:stack-id
+          Value: !Ref AWS::StackId
+  DovetailCdnArrangerBinaryLayer:
+    Type: AWS::Lambda::LayerVersion
+    Properties:
+      CompatibleRuntimes:
+        - nodejs14.x
+        - nodejs12.x
+      Content:
+        S3Bucket: prx-tech
+        S3Key: archives/ffmpeg-4.3.1-lambda-layer.zip
+      Description: ffmpeg binaries for dovetail-cdn-arranger
+      LicenseInfo: MIT
+  DovetailCdnArrangerExecutionIAMRole:
+    Type: 'AWS::IAM::Role'
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: 'Allow'
+            Principal:
+              Service:
+                - 'lambda.amazonaws.com'
+            Action:
+              - 'sts:AssumeRole'
+      Path: '/'
+      Policies:
+        - PolicyName: DynamoDBWritePolicy
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - 'dynamodb:DescribeTable'
+                  - 'dynamodb:GetItem'
+                  - 'dynamodb:PutItem'
+                Resource:
+                  - !Ref DynamodbTableArn
+        - PolicyName: S3WritePolicy
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - 's3:ListBucket'
+                Resource:
+                  - !GetAtt DovetailCdnArrangerWorkspaceBucket.Arn
+              - Effect: Allow
+                Action:
+                  - 's3:GetObject'
+                  - 's3:PutObject'
+                  - 's3:PutObjectTagging'
+                Resource:
+                  - !Sub ${DovetailCdnArrangerWorkspaceBucket.Arn}/*
+      ManagedPolicyArns:
+        - 'arn:aws:iam::aws:policy/CloudWatchLambdaInsightsExecutionRolePolicy'
+      Tags:
+        - Key: Project
+          Value: Dovetail
+        - Key: Environment
+          Value: !Ref EnvironmentType
+        - Key: 'prx:cloudformation:stack-name'
+          Value: !Ref AWS::StackName
+        - Key: 'prx:cloudformation:stack-id'
+          Value: !Ref AWS::StackId
+  DovetailCdnArrangerFunction:
+    Type: 'AWS::Lambda::Function'
+    Properties:
+      Code:
+        S3Bucket: !Ref CodeS3Bucket
+        S3Key: lambda/PRX-dovetail-cdn-arranger.zip
+        S3ObjectVersion: !Ref CodeS3ObjectVersion
+      Description: Dovetail CDN Arranger
+      Environment:
+        Variables:
+          DDB_TABLE: !Ref DynamodbTableArn
+          DOVETAIL_HOST: !Ref DovetailHost
```

Comment:
Is this the Dovetail _router_ host? If so, maybe the name could reflect that.

Reply:
It's actually either (they both have the API).  So purposefully left a bit ambiguous.